lib/app/app.rb

```diff
@@ -84,51 +85,92 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   post '/status' do
-    source_status = @connector.source_status(body_params)
+    connector = settings.connector_class.new
+
+    source_status = connector.source_status(body_params)
     json(
-      :extractor => { :name => @connector.name },
+      :extractor => { :name => connector.name },
       :contentProvider => source_status
     )
   end
 
+  post '/start_sync' do
+    job = settings.job_store.create_job
+
+    settings.job_runner.start_job(
+      job: job,
+      connector_class: settings.connector_class,
+      modified_since: body_params[:modified_since],
+      cursors: body_params[:cursors],
+      access_token: body_params[:access_token]
+    )
+
+    json(
+      :job_id => job.id,
+      :status => job.status
+    )
+  end
+
   post '/documents' do
-    results, cursors, completed = @connector.document_batch(body_params)
+    job = settings.job_store.fetch_job(body_params[:job_id])
```

Comment:
Should catch `JobNotFoundError`?

Reply:
I'm not sure about this - catching it will make us return something, and the design of the endpoint states that we return fields like "status", "docs" - we'll have to change the design a bit and I'm not sure we need to - this HTTP interface is a bit rudimentary and we won't support it really soon

Reply:
>catching it will make us return something

Exactly, we should tell the caller (i.e. ent-search) that we couldn't find the job ID. It's not necessary a 200 response, maybe 404. But here it will return 500 `INTERNAL SERVER ERROR` if we leave it as it is.

Reply:
Yeah, good call - I've fixed this to send 404s now: https://github.com/elastic/connectors/blob/375fa82fe12b94a2876abc68ca7eda8d681723c5/lib/app/app.rb#L130